plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java

```diff
@@ -298,7 +296,13 @@ public static Object convert(Object value, Type type)
             return ((Long) value) * PICOSECONDS_PER_MICROSECOND;
         }
         if (type instanceof Types.FloatType) {
-            return Float.floatToIntBits((Float) value);
+            return ((Float) value).longValue();
```

Comment:
this will cats to long and not preserve float. `Float.floatToIntBits((Float) value)` seems correct.

It seems that we didn't hit float case in test below? It it possible to test?